open3d_slam/include/open3d_slam/ScanToMapRegistration.hpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * ScanToMapRegistration.hpp
+ *
+ *  Created on: Oct 31, 2022
+ *      Author: jelavice
+ */
+
+#pragma once
+#include "open3d_slam/typedefs.hpp"
+#include "open3d_slam/Transform.hpp"
+#include "open3d_slam/Parameters.hpp"
+
+#include "open3d/pipelines/registration/Registration.h"
+
+#include <memory>
+
+namespace o3d_slam {
+
+class Submap;
+class CroppingVolume;
+
+using RegistrationResult = open3d::pipelines::registration::RegistrationResult;
+
+struct ProcessedScans {
+	PointCloudPtr merge_;
+	PointCloudPtr match_;
+};
+
+class ScanToMapRegistration {
+
+public:
+	ScanToMapRegistration() = default;
+	virtual ~ScanToMapRegistration() = default;
+	virtual ProcessedScans processForScanMatchingAndMerging(const PointCloud &in,
+			const Transform &mapToRangeSensor) const =0;
+	virtual RegistrationResult scanToMapRegistration(const PointCloud &scan, const Submap &activeSubmap,
+			const Transform &mapToRangeSensor, const Transform &initialGuess) const = 0;
+};
+
+class ScanToMapIcpOpen3D : public ScanToMapRegistration {
```

Comment:
ICP seems to be still fairly general for implementing the scanToMapRegistration already. Don't you wanna split it up in the different ICP types at this level?

Reply:
I did it, but it hasn't been pushed yet. WAnted to keep this PR reasonably sized.